plugins/wasi_nn/piper.cpp

```diff
@@ -0,0 +1,487 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+#include "piper.h"
+#include "wasinnenv.h"
+
+#ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_PIPER
+#include "simdjson.h"
+#include "types.h"
+#include <cstring>
+#include <fstream>
+#include <functional>
+#include <ios>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <optional>
+#include <piper.hpp>
+#include <string>
+#include <string_view>
+#include <vector>
+#endif
+
+namespace WasmEdge::Host::WASINN::Piper {
+#ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_PIPER
+
+template <typename T, typename U = T>
+Expect<WASINN::ErrNo> getOptionalOption(simdjson::dom::element &Doc,
+                                        std::string_view Key,
+                                        std::optional<T> &Result) {
+  if (Doc.at_key(Key).error() == simdjson::SUCCESS) {
+    auto Value = U{};
+    auto Err = Doc[Key].get<U>().get(Value);
+    if (Err) {
+      spdlog::error(
+          "[WASI-NN] Piper backend: Unable to retrieve the {} option.", Key);
```

Comment:
Let's apply `string_view_literal` for these error messages. Use `"xxx"sv` instead.